modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue

```diff
@@ -34,7 +34,10 @@
             <div class="apos-doc-editor__body">
               <AposSchema
                 v-if="docReady"
-                :schema="schema" :current-fields="currentFields"
+                :key="'schema-body-validation-' + triggerValidation"
```

Comment:
When the `triggerValidation` (a boolean) is activated (after clicking on "submit"), the key changes, and this forces the schema to re-render its sub-components, forcing them to validate the value they hold.

Reply:
OK, but why not just pass it as a boolean prop?

Reply:
Not sure why the string concatenation.

Reply:
Did you mean having `:key="triggerValidation"`? That means two components would have the same key in `AposDocEditor` because there are 2 `AposSchema`s.  I don't know if this would be a problem.

Reply:
This feels like tricking the schema into validating rather than explicitly telling it to.

Reply:
`triggerValidation` was not enough to really trigger the validation because the component has already been mounted. It won't validate until it's updated and Vue won't consider the component is updated by only changing the value of `triggerValidation` but it will if the key is changed. So, yes, it is tricky. But I did not find a good alternative. Unless you have a solution in mind.